src/Kunstmaan/MultiDomainBundle/Controller/SiteSwitchController.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Kunstmaan\MultiDomainBundle\Controller;
+
+use Kunstmaan\MultiDomainBundle\Helper\DomainConfiguration;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+class SiteSwitchController extends Controller
+{
+    /**
+     * @Route("/switch-site", name="KunstmaanMultiDomainBundle_switch_site")
+     * @Method({"GET"})
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return Response
+     * @throws AccessDeniedException
+     * @throws NotFoundHttpException
+     */
+    public function switchAction(Request $request)
+    {
+        if (!$this->get('security.context')->isGranted('ROLE_SUPER_ADMIN')) {
+            throw $this->createAccessDeniedException();
+        }
+
+        $domainConfiguration = $this->get('kunstmaan_node.domain_configuration');
+        $host = $request->query->get('host');
+        $hosts = $domainConfiguration->getHosts();
+        if (!in_array($host, $hosts)) {
+            throw $this->createNotFoundException('Invalid host specified');
+        }
+
+        $this->get('session')->set(DomainConfiguration::OVERRIDE_HOST, $host);
```

Comment:
Maybe show a big warning in the backend that you are showing another domain?, with a return button to make the override empty again?